homeassistant/components/islamic_prayer_times/sensor.py

```diff
@@ -56,10 +58,22 @@ async def async_setup_entry(
 
     coordinator: IslamicPrayerDataUpdateCoordinator = hass.data[DOMAIN]
 
-    async_add_entities(
-        IslamicPrayerTimeSensor(coordinator, description)
-        for description in SENSOR_TYPES
-    )
+    entities: list[IslamicPrayerTimeSensor] = []
+
+    # Add entry_id to the unique_id to allow multiple config entries
+    entity_registry = er.async_get(hass)
+
+    for description in SENSOR_TYPES:
+        old_unique_id = description.key
+        if entity_id := entity_registry.async_get_entity_id(
+            PLATFORM, DOMAIN, old_unique_id
+        ):
+            new_unique_id = f"{config_entry.entry_id}-{description.key}"
+            entity_registry.async_update_entity(entity_id, new_unique_id=new_unique_id)
```

Comment:
This is no good, if there are multiple config entries, they will now fight over the original entities.
Please use the function `entity_registry.async_migrate_entries` instead which is specifically made for updating unique_ids for entities which belong to a given config entry.